src/vs/platform/extensionManagement/node/extensionGalleryService.ts

```diff
@@ -204,6 +206,13 @@ function getStatistic(statistics: IRawGalleryExtensionStatistics[], name: string
 function getVersionAsset(version: IRawGalleryExtensionVersion, type: string): IGalleryExtensionAsset {
 	const result = version.files.filter(f => f.assetType === type)[0];
 
+	// {{SQL CARBON EDIT}}
+	let properties = version.properties.filter(p => p.key === type);
+	let uriFromProperty: string = undefined;
+	if (properties && properties.length >= 1) {
+		uriFromProperty = properties[0].value;
```

Comment:
Looks like this works but I'm not sure I understand how. Why is the uri always `properties[0]`?

Reply:
technically we can set the property with key  "Microsoft.SQLOps.DownloadPage" more than once but I assume here if it's set, only the first value is valid and it's the uri

---
In reply to: [168323032](https://github.com/Microsoft/sqlopsstudio/pull/689#discussion_r168323032) [](ancestors = 168323032)